app.py

```diff
@@ -1,72 +1,179 @@
-#!/bin/env python
+#!/usr/bin/env python
 """
 This example uses docopt with the built in cmd module to demonstrate an
 interactive command application.
+
 Usage:
-    dojo create_room <room_type> <room_names> ...
-    dojo add_room (<fname> <lname> <FELLOW/STAFF> [<wants_accommodation>])
-    dojo [-i | --interactive]
-    dojo
-    dojo (-h | --help | --version)
+    app.py
+    app.py create_room <room_type> <room_names>...
+    app.py add_person <fname> <lname> <FELLOW/STAFF> [<wants_accommodation>]
+    app.py print_room <room_name>
+    app.py print_allocations [(-o <filename>)]
+    app.py print_unallocated [(-o <filename>)]
+    app.py people_id
+    app.py reallocate_person <person_identifier> <room_name>
+    app.py load_people
+    app.py save_state [(--db sqlite_database)]
+    app.py load_state <sqlite_database>
+    app.py (-i | --interactive)
+    app.py (-h | --help | --version)
+
 Options:
     -i, --interactive  Interactive Mode
     -h, --help  Show this screen and exit.
 """
 
-#import necessary modules
 import sys
 import cmd
 from docopt import docopt, DocoptExit
 
+from dojo import Dojo
+
+
+def docopt_cmd(func):
+    """
+    This decorator is used to simplify the try/except block and pass the result
+    of the docopt parsing to the called action.
+    """
+
+    def fn(self, arg):
+        try:
+            opt = docopt(fn.__doc__, arg)
+
+        except DocoptExit as e:
+            # The DocoptExit is thrown when the args do not match.
+            # We print a message to the user and the usage block.
+
+            print('Invalid Command!')
+            print(e)
+            return
+
+        except SystemExit:
+            # The SystemExit exception prints the usage for --help
+            # We do not need to do the print here.
+
+            return
+
+        return func(self, opt)
 
+    fn.__name__ = func.__name__
+    fn.__doc__ = func.__doc__
+    fn.__dict__.update(func.__dict__)
+    return fn
 
 
-#for interactive shell define the class
-class MyInteractive (cmd.Cmd):
+class MyInteractive(cmd.Cmd):
     intro = 'Welcome to my interactive program!' \
         + ' (type help for a list of commands.)'
     prompt = 'dojo>>>'
     file = None
 
-    #use decorators to define the functions
     @docopt_cmd
     def do_create_room(self, arg):
```

Comment:
I cringe anytime I see arg considering it holds multiple arguments I believe it should be more aptly named something plural e.g. args. This applies everywhere you use this variable name